### PR TITLE
Smooth difference grid

### PR DESCRIPTION
**Pull request by JamieJQuinn**
Currently remove-restore applies a difference grid to the base grid directly. This presents an issue at the boundary of the difference grid (for example, where there is no high-resolution data in a region) and creates a discontinuity between high and low resolution regions (see #32). We have already tried:

- Gridding the difference grid with pygmt's `surface` instead of `nearneighbour` however this introduces unacceptably large changes to the base grid far from the region being updated, regardless of tension parameter.
- Filtering the difference grid prior to application to the base grid, or filtering the base grid after updating, however this doesn't adequately solve the problem and smooths the entire domain, reducing accuracy away from the problematic boundaries.

The solution which appears to work well is windowing. A border of a certain width is created around the boundary of the difference grid and the data within that boundary smoothed. This softens the boundary between the difference grid and the base grid and removes the hard discontinuity. 

The actual way in which this is implemented is as follows:
- the difference data is gridded using nearneighbour and empty cells set to 9999, representing no data
- a grid is created which has values 1.0 where there is data and 0.0 where there is none
- this grid is filtered using `grdfilter` to create a window either side of the no-data boundary
- the grid is scaled and shifted so that exactly on the boundary the grid has a value of 0.0, in the interior of the difference grid the value is 1.0 and there is a linear interpolation between, inside the boundary region. All points beyond the boundary, where there is no data, are set to 0.
- this grid is applied as a filter (really just a multiplication) to the difference grid to remove the no data values and apply the window along the boundary.

### CODE REVIEW

**Comment by JamieJQuinn** [comment]
Just for testing, there is a clear discontinuity when using:
```
remove-restore --base ../bath_data/GEBCO_2019_-157.0_62.0_-121.0_35.0.nc --spacing 0.001 --output test.nc ../bath_data/sample_data.tif --region_of_interest -123.3 -122.8 48.800 49.00 --plot
```
which is alleviated when adding the `--window_width` flag with a value 10 times the output grid spacing:
```
remove-restore --base ../bath_data/GEBCO_2019_-157.0_62.0_-121.0_35.0.nc --spacing 0.001 --output test.nc ../bath_data/sample_data.tif --region_of_interest -123.3 -122.8 48.800 49.00 --window_width 0.05 --plot
```

File: pycascadia/remove_restore.py [hunk 1]
```diff
@@ -68,13 +83,16 @@ def main():
     parser.add_argument('--output', required=True, help='filename of final output')
     parser.add_argument('--region_of_interest', required=False, nargs=4, type=float,
                         help='output region in order <xmin> <xmax> <ymin> <ymax>. Defaults to the extent of the base grid.')
+    parser.add_argument('--window_width', required=False, type=float,
+                        help='Enable windowing of update gride and specify width of window in degrees')
```
Comment (alessandrofelder) [inline pycascadia/remove_restore.py:87]:
```suggestion
                        help='Enable windowing of update grid and specify width of window in degrees')
```
small typo.

**Review by alessandrofelder — CHANGES_REQUESTED** [review]
Looking good - (manual) test runs you showed me are very rewarding to look at :) 
I think the windowing approach would benefit from a couple of comments (but happy to be contradicted of course).

File: pycascadia/remove_restore.py (hunk 1 above)
Comment (JamieJQuinn) [reply pycascadia/remove_restore.py:87]: You don'ta like ma Italian?

**Review by alessandrofelder — APPROVED** [review]
👍

**Comment by alessandrofelder** [comment]
The one test failure looks like a connectivity issue, I'm not worried...

**Pull request merged**